Lib/test/test_capi/test_opt.py

```diff
@@ -775,10 +775,13 @@ def testfunc(n):
             a = 1.0
             for _ in range(n):
                 a = a + 0.1
+                a = a + 0.1
+                a = a + 0.1
+                a = a + 0.1
             return a
 
         res, ex = self._run_with_optimizer(testfunc, 32)
-        self.assertAlmostEqual(res, 4.2)
+        self.assertAlmostEqual(res, 13.8)
```

Comment:
Could you change all the float examples to use fractional parts that are powers of 2? (E.g. 0.5, 0.25, 0.125.) Adding 0.1 + 0.1 + 0.1 + 0.1 and expecting the outcome to be 0.4 is just not safe, due to float rounding and decimal/binary issues.